subworkflows/nf-core/fasta_index_bismark_bwameth/main.nf

```diff
@@ -81,10 +83,10 @@ workflow FASTA_INDEX_BISMARK_BWAMETH {
     /*
     * Generate fasta index if not supplied for bwameth workflow or picard collecthsmetrics tool
     */
-    if (params.aligner == 'bwameth' | params.run_picard_collecthsmetrics) {
+    if (aligner == 'bwameth' | collecthsmetrics) {
         // already exising fasta index
         if (fasta_index) {
-            ch_fasta_index = Channel.value(file(fasta_index, checkIfExists: true))
+            ch_fasta_index = Channel.value(file(fasta_index, checkIfExists: true)).map{ index -> [ [:], index ]}
```

Comment:
Why is this making a channel, it should already be an channel from the input?

Reply:
but, we want to cast the index as a value channel. u suggesting use to use .first() or something ?

Reply:
@SPPearce I tried to move the value channel creation out of this sub-wf to the main but, wasn't getting an easy and elegant solution. If you can let this pass this time, I will refactor the connections in a next release. lmk ur thoughts

Reply:
@SPPearce let me thoughts on above plz.